mypy/checker.py

```diff
@@ -1234,13 +1235,16 @@ def check_func_def(
                             new_frame.types[key] = narrowed_type
                             self.binder.declarations[key] = old_binder.declarations[key]
                 with self.scope.push_function(defn):
-                    # We suppress reachability warnings when we use TypeVars with value
+                    # We suppress reachability warnings for empty generators (return; yield), since there's
+                    # no way to promote a function into a generator except by adding an "unreachable" yield.
```

Comment:
I still think it would be more precise to refer to "empty generator functions" (functions with nearly-empty bodies that return generators) here rather than "empty generators" (this _is_ mypy after all ;)

```suggestion
                    # We suppress reachability warnings for empty generator functions
                    # (return; yield), since the most idiomatic way to promote a function into a
                    # generator function is often to add an "unreachable" yield.
```